load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -15,124 +15,222 @@
+import com.beust.jcommander.ParameterException;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Timestamp;
+import com.google.pubsub.flic.common.LatencyDistribution;
+import com.google.pubsub.flic.controllers.Client;
+import com.google.pubsub.flic.controllers.Client.ClientType;
+import com.google.pubsub.flic.controllers.ClientParams;
+import com.google.pubsub.flic.controllers.Controller;
+import com.google.pubsub.flic.controllers.GCEController;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Drives the execution of the framework through command line arguments. */
-public class Driver {
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.LogManager;
+import java.util.stream.LongStream;
 
-  private static final Logger log = LoggerFactory.getLogger(Driver.class.getName());
+/**
+ * Drives the execution of the framework through command line arguments.
+ */
+class Driver {
+  private final static Logger log = LoggerFactory.getLogger(Driver.class);
+  @Parameter(
+      names = {"--help"},
+      help = true
+  )
+  private boolean help = false;
+  @Parameter(
+      names = {"--cps_publisher_count"},
+      description = "Number of CPS publishers to start."
+  )
+  private int cpsPublisherCount = 1;
+  @Parameter(
+      names = {"--cps_subscriber_count"},
+      description = "Number of CPS subscribers to start."
+  )
+  private int cpsSubscriberCount = 1;
+  @Parameter(
+      names = {"--kafka_publisher_count"},
+      description = "Number of Kafka publishers to start."
+  )
+  private int kafkaPublisherCount = 0;
+  @Parameter(
+      names = {"--kafka_subscriber_count"},
+      description = "Number of Kafka subscribers to start."
+  )
+  private int kafkaSubscriberCount = 0;
+  @Parameter(
+      names = {"--message_size", "-m"},
+      description = "Message size in bytes (only when publishing messages).",
+      validateWith = GreaterThanZeroValidator.class
+  )
+  private int messageSize = 100000;
+  @Parameter(
+      names = {"--loadtest_seconds"},
+      description = "Duration of the load test, in seconds.",
+      validateWith = GreaterThanZeroValidator.class
+  )
+  private int loadtestLengthSeconds = 60;
+  @Parameter(
+      names = {"--project"},
+      required = true,
+      description = "Google Cloud Platform project name."
+  )
+  private String project = "";
+  @Parameter(
+      names = {"--batch_size", "-b"},
+      description = "Number of messages to batch per publish request.",
+      validateWith = GreaterThanZeroValidator.class
+  )
+  private int batchSize = 10;
+  @Parameter(
+      names = {"--subscription_fanout"},
+      description = "Number of subscriptions to create for each topic. Must be at least 1.",
```

Comment:
Done. Multiple topics will be used when we have multiple clients. When we want to load test the Python library vs the Java library each will use a separate topic. But this is CPS only.